09-19/backend-update-and-delete_products/components/Product/index.js

```diff
@@ -0,0 +1,85 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import { ProductCard } from "./Product.styled";
+import { StyledLink } from "../Link/Link.styled";
+import Comments from "../Comments";
+import { useState } from "react";
+import ProductForm from "../ProductForm";
+
+export default function Product() {
+  const router = useRouter();
+  const [isEditMode, setIsEditMode] = useState(false);
+  const { id } = router.query;
+
+  const { data, isLoading, mutate } = useSWR(`/api/products/${id}`);
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!data) {
+    return;
+  }
+
+  async function handleEditProduct(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const productData = Object.fromEntries(formData);
+
+    const response = await fetch(`/api/products/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(productData),
+    });
+
+    if (response.ok) {
+      mutate();
+    }
+
+    return;
+  }
+
+  async function handleDeleteProduct(id) {
```

Comment:
its not necessary to use an input here, the id is accessbile to the function because its in the scope of the component.